subprojects/file-watching/src/test/groovy/org/gradle/internal/watch/registry/impl/WatchedHierarchiesTest.groovy

```diff
@@ -16,56 +16,42 @@
 
 package org.gradle.internal.watch.registry.impl
 
-import org.gradle.util.Requires
-import org.gradle.util.TestPrecondition
+import org.gradle.internal.snapshot.TestSnapshotFixture
 import spock.lang.Specification
 
-import java.nio.file.Paths
-
-class WatchedHierarchiesTest extends Specification {
-
-    @Requires(TestPrecondition.UNIX_DERIVATIVE)
-    def "resolves recursive UNIX roots #directories to #resolvedRoots"() {
-        expect:
-        resolveRecursiveRoots(directories) == resolvedRoots
-
-        where:
-        directories          | resolvedRoots
-        []                   | []
-        ["/a"]               | ["/a"]
-        ["/a", "/b"]         | ["/a", "/b"]
-        ["/a", "/a/b"]       | ["/a"]
-        ["/a/b", "/a"]       | ["/a"]
-        ["/a", "/a/b/c/d"]   | ["/a"]
-        ["/a/b/c/d", "/a"]   | ["/a"]
-        ["/a", "/b/a"]       | ["/a", "/b/a"]
-        ["/b/a", "/a"]       | ["/a", "/b/a"]
-        ["/a", "/b/a", "/a"] | ["/a", "/b/a"]
+class WatchedHierarchiesTest extends Specification implements TestSnapshotFixture {
+    def "does not watch when there's nothing to watch"() {
+        def watchable = Mock(WatchableHierarchies)
+        when:
+        def watched = WatchedHierarchies.resolveWatchedHierarchies(watchable, buildHierarchy([]))
+        then:
+        watched.watchedRoots as List == []
+        1 * watchable.recentlyUsedHierarchies >> []
     }
 
-    @Requires(TestPrecondition.WINDOWS)
-    def "resolves recursive Windows roots #directories to #resolvedRoots"() {
-        expect:
-        resolveRecursiveRoots(directories) == resolvedRoots
-
-        where:
-        directories                 | resolvedRoots
-        []                             | []
-        ["C:\\a"]                      | ["C:\\a"]
-        ["C:\\a", "C:\\b"]             | ["C:\\a", "C:\\b"]
-        ["C:\\a", "C:\\a\\b"]          | ["C:\\a"]
-        ["C:\\a\\b", "C:\\a"]          | ["C:\\a"]
-        ["C:\\a", "C:\\a\\b\\c\\d"]    | ["C:\\a"]
-        ["C:\\a\\b\\c\\d", "C:\\a"]    | ["C:\\a"]
-        ["C:\\a", "C:\\b\\a"]          | ["C:\\a", "C:\\b\\a"]
-        ["C:\\b\\a", "C:\\a"]          | ["C:\\a", "C:\\b\\a"]
-        ["C:\\a", "C:\\b\\a", "C:\\a"] | ["C:\\a", "C:\\b\\a"]
+    def "watches empty directory"() {
+        def watchable = Mock(WatchableHierarchies)
+        def dir = new File("/empty")
```

Comment:
```suggestion
        def dir = new File("empty").absoluteFile
```

Instead of using these absolute paths in the `File` constructor which may end up to be not absolute on Windows,  how about using a relative path with `.absoluteFile`?

Reply:
OK.